wandb/sdk/wandb_run.py

```diff
@@ -1854,77 +1888,118 @@ def save(
             )
             return True
 
-        return self._save(glob_str, base_path, policy)
+        if isinstance(glob_str, bytes):
+            # Preserved for backward compatibility: allow bytes inputs.
+            glob_str = glob_str.decode("utf-8")
+        elif isinstance(glob_str, str) and (
+            glob_str.startswith("gs://") or glob_str.startswith("s3://")
+        ):
+            # Provide a better error message for a common misuse.
+            wandb.termlog(
+                f"{glob_str} is a cloud storage url, can't save file to wandb."
+            )
+            return []
+        glob_path = pathlib.Path(glob_str)
+
+        if base_path is not None:
+            base_path = pathlib.Path(base_path)
+        elif not glob_path.is_absolute():
+            base_path = pathlib.Path(".")
+        else:
+            # Absolute glob paths with no base path get special handling.
+            wandb.termwarn(
+                "Saving files without folders. If you want to preserve "
+                "subdirectories pass base_path to wandb.save, i.e. "
+                'wandb.save("/mnt/folder/file.h5", base_path="/mnt")',
+                repeat=False,
+            )
+            base_path = glob_path.resolve().parent.parent
```

Comment:
why 2 levels?

Reply:
so that if glob_path is "/Users/timoffex/Documents/files/file.txt", then base_path is "/Users/timoffex/Documents", and the file is saved as "files/file.txt"

Reply:
(I'm trying to preserve the original semantics, but IMO it would be cleaner to save the file as "file.txt")

Reply:
oh yuk 😭